### PR TITLE
Adding EventHub lifecycle tests;include necessary depencency

### PR DESCRIPTION
**Pull request by aljannuzzi**
Relates to #691

@zhongyi-Zhang: Added a function inside the lifecycle test for EventHub and added the appropriate dependency.

### CODE REVIEW

**Comment by zhongyi-zhang** [comment]
It looks the vendor verification failed. Could you run `make dep` to fix it?

**Comment by aljannuzzi** [comment]
@zhongyi-zhang : I formatted the imports, fixed some other lint warnings and included the Gopkg.lock file to pass the dep ensure. "make dep" worked for me but I had forgotten to include the lock file. Thanks!

**Comment by zhongyi-zhang** [comment]
@aljannuzzi thanks for following up. Could you fix the lint issues according to the linked "Details"? The `make lint` should work locally, too.

**Comment by aljannuzzi** [comment]
Hi @zhongyi-zhang , thanks for your guidance. Quick comment: I noticed that in the line 133 in the Makefile, GOPATH is being set as /tmp$$GOPATH. Shouldn't it be /tmp$$PRJ_DIR?

I suspect "dep ensure" is not being executed because of that and it is making the vendored code test fail and all the others. As I've run locally make dep, I have it working from my side.

I will add the changaed make file as part of the commit as well so you can evaluate.

Thanks again!

File: tests/lifecycle/eventhub_cases_test.go [hunk 1]
```diff
@@ -1,7 +1,13 @@
-// +build !unit
```
Comment (zhongyi-zhang) [inline tests/lifecycle/eventhub_cases_test.go]: The unit test failed for this. It should not be removed as the lifecycle test is not unit test.

Comment (aljannuzzi) [reply tests/lifecycle/eventhub_cases_test.go]: I returned the comment. Sorry, it was accidentally removed while cleaning other comments.

Comment (zhongyi-zhang) [reply tests/lifecycle/eventhub_cases_test.go]: no problem:)

**Comment by zhongyi-zhang** [comment]
Oh, quay.io is down... Our base image is there. Ci blocked now :(.

**Comment by aljannuzzi** [comment]
@zhongyi-zhang : do I need to submitt it again or just wait for the jobs to be re-scheduled? Thanks!

File: tests/lifecycle/eventhub_cases_test.go [hunk 2]
```diff
@@ -14,5 +22,43 @@ var eventhubsTestCases = []serviceLifecycleTestCase{
...
+	connStr := connStrP1 + ";EntityPath=" + hubName
+	hub, err := eventhub.NewHubFromConnectionString(connStr)
+	if err != nil {
+		return fmt.Errorf("error creating eventhub client object: %s", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 80*time.Second)
+	defer cancel()
+	err = hub.Send(ctx, eventhub.NewEventFromString("hello, world!"))
+	if err != nil {
+		return fmt.Errorf("error sending message to eventhub: %s", err)
```
Comment (zhongyi-zhang) [inline tests/lifecycle/eventhub_cases_test.go]: The latest lifecycle test failed here:
```
--- FAIL: TestServices (3449.00s)
    --- FAIL: TestServices/lifecycle (0.00s)
        --- FAIL: TestServices/lifecycle/TestServices/lifecycle/eventhubs/eventhubs (63.60s)
	Error Trace:	driver_test.go:50
	Error:      	Expected nil, but got: &errors.errorString{s:"error sending message to eventhub: amqp: receiver settlement mode \"second\" requested, received \"first\" from server"}
```

And a similar issue in the client repo: https://github.com/Azure/azure-event-hubs-go/issues/101

Comment (aljannuzzi) [reply tests/lifecycle/eventhub_cases_test.go]: Is it possible to run the lifecycle tests for EH in the eastus datacenter? I created a simple send/receive function and it works on eastus but gets the same error in southcentral.

Comment (devigned) [reply tests/lifecycle/eventhub_cases_test.go]: This error should be fixed in https://github.com/Azure/azure-event-hubs-go/pull/104.

Please open an issue in EventHubs if you run into this issue again.

Comment (zhongyi-zhang) [reply tests/lifecycle/eventhub_cases_test.go]: Thanks @devigned for the fix! @aljannuzzi it looks changing datacenter works, too. I'm going to merge this PR. Thanks for adding the test case!

**Pull request merged**